### PR TITLE
Modified commands

### LINKED ISSUE #15: [good-first-issue] Respond to commands only when mentioned
**Summary**
Right now the bot responds to any command followed by `/`. This may create conflicts if there are multiple bots in the group and a
given command is not meant for this bot.
So the bot should respond only when it is addressed. E.g: `/command@bot_name`

**Implementation**
A possible way to implement this is using regular expressions. We can add regex filters in `CommandHandler`s.
https://github.com/JIITODC/bot/blob/22302be3e738a49ec22da6be882a50f16376a292/bot.py#L54
Here is a brief documentation on the regex filter that the `python-telegram-bot` api wrapper provides:
https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.filters.html#telegram.ext.filters.Filters.regex 

Lastly, a little guide on regular expressions:
https://medium.com/factory-mind/regex-tutorial-a-simple-cheatsheet-by-examples-649dc1c3f285

**Alternatives**
Do suggest in the comments if you have some other approach in mind. :smiley:

**If this is your first time contributing**
Hey! :wave: If you're interested to work on this issue, do comment, "I would like to work on this issue. Please assign this to me.", below to let others know that you're interested in working on this. Also, do take a look at the [contributing guide](CONTRIBUTING.md) which will take you step by step from forking this repo to creating a pull request :rocket: :fire:. 
We use the `python-telegram-bot` api wrapper. Here is a link to its documentation: https://python-telegram-bot.readthedocs.io/en/stable/index.html
Feel free to ask anything related to this issue :hugs:

### PR DESCRIPTION
**Pull request by Abjcodes**
## Description
 Updated commands to respond only to /command@bot_name by using Regexhandler and regular expressions.

## Does this resolve any currently open issues?
Resolves #15 

## Screenshot/GIF showing resolved bug/enhancement (if applicable)
..
Screenshot_20201112_111319_org telegram messenger

## Checklist
Just a friendly reminder to update dependencies/documentation if required. Do checkout if applicable.
- [ ] Updated the [requirements.txt](requirements.txt) (new dependency)
- [ ] Updated the [documentation](README.md) (setup instructions changed)

## Additional points worth mentioning
Updated commands/help.py for showing modified commands.

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -51,11 +51,11 @@ def check(update, context):
...
-    dp.add_handler(CommandHandler("help", help))
-    dp.add_handler(CommandHandler("xkcd", xkcd))
-    dp.add_handler(CommandHandler("links", links))
-    dp.add_handler(CommandHandler("set_meetup", set_meetup))
-    dp.add_handler(CommandHandler("meetup", meetup))
+    dp.add_handler(MessageHandler(Filters.regex(r'^(/help@%s)$' % bot_name), help))
+    dp.add_handler(MessageHandler(Filters.regex(r'^(/xkcd@%s)$' % bot_name), xkcd))
+    dp.add_handler(MessageHandler(Filters.regex(r'^(/links@%s)$' % bot_name), links))
+    dp.add_handler(MessageHandler(Filters.regex(r'^(/set_meetup@%s)$' % bot_name), set_meetup))
+    dp.add_handler(MessageHandler(Filters.regex(r'^(/meetup@%s)$' % bot_name), meetup))
```
Comment (DelusionalOptimist) [inline bot.py:62]: Do we need to change the handlers from `CommandHandler` to `MessageHandler`? :thinking: 
Ig we can use `Filters` with `CommandHandler` too.

**Review by DelusionalOptimist — CHANGES_REQUESTED** [review]
Hey @Abjcodes :wave: !
Great work! :fire: 

As I see, the bot still sends `Sorry, I don't know that command` when a `/command` is given. Can you do something about this?

File: bot.py (hunk 1 above)
Comment (Abjcodes) [reply bot.py:62]: Yeah, I have tried using `CommandHandler ` but it's giving "regex is not iterable" error. I think `MessageHandler` will be fine for now.

**Comment by Abjcodes** [comment]
> Hey @Abjcodes 👋 !
> Great work! 🔥
> 
> As I see, the bot still sends `Sorry, I don't know that command` when a `/command` is given. Can you do something about this?

Sure, I will look into it.

**Comment by aayushmau5** [comment]
@Abjcodes still working on it ?

**Comment by Abjcodes** [comment]
> @Abjcodes still working on it ?

Sorry for the delay, I will raise a PR within 3-4 days as I am caught in the middle of a couple of things. Will that be ok?

**Comment by aayushmau5** [comment]
No problem. Take your time. :)

**Comment by Abjcodes** [comment]
@aayushmau5 @DelusionalOptimist I have made the necessary changes. Please do notify me if there are more changes to be made.

**Review by DelusionalOptimist — APPROVED** [review]
LGTM. Thanks @Abjcodes :fire:

Feel free to checkout other [issues](https://github.com/JIITODC/bot/issues) too. :grin:

**Comment by Abjcodes** [comment]
> LGTM. Thanks @Abjcodes :fire:
> 
> Feel free to checkout other [issues](https://github.com/JIITODC/bot/issues) too. :grin:  

Sure :)

**Review by aaryanporwal — APPROVED** [review]

**Pull request merged**